### PR TITLE
Feature 5024 extend list dataset replicas bulk to use the   deep flag

### PR DESCRIPTION
**Pull request by ivmfnal**

### CODE REVIEW

File: lib/rucio/api/replica.py [hunk 1]
```diff
@@ -80,29 +81,17 @@ def declare_bad_file_replicas(pfns, reason, issuer, vo='def'):
     """
     Declare a list of bad replicas.
 
-    :param pfns: Either a list of PFNs (string) or a list of replicas {'scope': <scope>, 'name': <name>, 'rse_id': <rse_id>}.
+    :param pfns: The list of PFNs.
```
Comment (bari12) [inline lib/rucio/api/replica.py:84]: Please add the changes to bad_replicas in a different PR, this shouldn't be folded in with the dataset replicas one.

File: lib/rucio/core/replica.py [hunk 2]
```diff
@@ -41,10 +41,12 @@
 # - David Población Criado <david.poblacion.criado@cern.ch>, 2021
 # - Joel Dierkes <joel.dierkes@cern.ch>, 2021-2022
 # - Christoph Ames <christoph.ames@physik.uni-muenchen.de>, 2021
+# - Igor Mandrichenko <ivm@fnal.gov>, 2022
+
+from __future__ import print_function
```
Comment (bari12) [inline lib/rucio/core/replica.py:46]: Why? This is already py3 code

File: lib/rucio/core/replica.py [hunk 3]
```diff
@@ -85,11 +86,9 @@
                                      BadFilesStatus, RuleState, BadPFNStatus)
 from rucio.db.sqla.session import (read_session, stream_session, transactional_session,
                                    DEFAULT_SCHEMA_NAME, BASE)
-from rucio.db.sqla.types import InternalScopeString, String
-from rucio.db.sqla.util import create_temp_table
 from rucio.rse import rsemanager as rsemgr
 
-REGION = make_region_memcached(expiration_time=60)
+REGION = make_region().configure('dogpile.cache.memory', expiration_time=60)
```
Comment (bari12) [inline lib/rucio/core/replica.py:91]: Why bypass memcache?

File: lib/rucio/core/replica.py [hunk 4]
```diff
@@ -147,74 +146,40 @@ def get_bad_replicas_summary(rse_expression=None, from_date=None, to_date=None,
 
 
 @read_session
-def __exist_replicas(rse_id, replicas, session=None):
+def __exists_replicas(rse_id, scope=None, name=None, path=None, session=None):
```
Comment (bari12) [inline lib/rucio/core/replica.py:149]: Again, please take this out and do it in a different PR not intermixed with the deep one.

File: lib/rucio/core/replica.py [hunk 5]
```diff
@@ -347,7 +312,7 @@ def __declare_bad_file_replicas(pfns, rse_id, reason, issuer, status=BadFilesSta
     """
     Declare a list of bad replicas.
 
-    :param pfns: Either a list of PFNs (string) or a list of replicas {'scope': <scope>, 'name': <name>, 'rse_id': <rse_id>}.
+    :param pfns: The list of PFNs.
```
Comment (bari12) [inline lib/rucio/core/replica.py:315]: Same

File: lib/rucio/core/replica.py [hunk 6]
```diff
@@ -2056,142 +2023,6 @@ def list_and_mark_unlocked_replicas(limit, bytes_=None, rse_id=None, delay_secon
     """
     List RSE File replicas with no locks.
 
-    :param limit:                    Number of replicas returned.
```
Comment (bari12) [inline lib/rucio/core/replica.py:2059]: Why is this being removed? It feels like you are accidently overwriting Radus changes here.

**Review by bari12 — CHANGES_REQUESTED** [review]
Hi @ivmfnal I think there is something quite wrong with this PR. You are removing changes from other recent PRs and everything seems to be quite mixed up. E.g. please don't intermix bad_replica changes with the --deep changes, this should be two different PRs. 
Thanks

**Comment by ivmfnal** [comment]
yes I am going to remove this PR and start all over when the dust settles

**Comment by ivmfnal** [comment]
Removing the PR because someone else is modifying same files. Will wait for "quiet" time
